terraform/batch-processing/src/batch_processing.py

```diff
@@ -0,0 +1,27 @@
+import os
+import tempfile
+from time import time
+from boto3 import resource
+
+
+def lambda_handler(_event, _context):
+    source_bucket_name = _event.get("Records")[0].get("s3").get("bucket").get("name")
+    dest_bucket_name = source_bucket_name.replace("source", "destination")
+    output_bucket = resource('s3').Bucket(dest_bucket_name)
+
+    # Create a temporary dir to write our output file to, this avoids polluting /tmp with test files.
+    # Using the context manager here ensures the temp dir is cleaned up when the with statement is closed.
+    with tempfile.TemporaryDirectory() as tmpdir:
+        os.chdir(tmpdir)
+
+        # Write some placeholder bytestring data to a file in the bucket,
+        # so we can test that the lambda writes to the correct output bucket.
+        filename = f"output_report_{time()}.txt"
+        data = (b'Test file to see if the lambda writes to the correct s3 bucket. '
+                b'If our AWS bill skyrockets, this file has been written to the wrong bucket!')
+
+        output_bucket.put_object(Body=data, Key=filename)
```

Comment:
Why we are creating that tmp directory? When we all the actual logic all the data is in memory (it's based on response data). Is it necessary to create a file?